### ISSUE TITLE
cannot reschedule when propagationpolicy changed

### ISSUE THREAD

**Issue by merryzhou** [karmada-io/karmada#1411]
**What happened**:
1. two clusters: member1 and member2, both in ready status
2. a deployment with 2 replicas, and the propagationpolicy spec is:
```
spec:
  placement:
    clusterAffinity:
      clusterNames:
      - member1
    replicaScheduling:
      replicaDivisionPreference: Weighted
      replicaSchedulingType: Divided
      weightPreference:
        dynamicWeight: AvailableReplicas
```
3. wait pods of deployment deployed to member1. then update propagationpolicy to:
```
spec:
  placement:
    clusterAffinity:
      clusterNames:
      - member2                               # this line changed
    replicaScheduling:
      replicaDivisionPreference: Weighted
      replicaSchedulingType: Divided
      weightPreference:
        dynamicWeight: AvailableReplicas
```

i hope the deployment can be moved to member2 cluster, but nothing happend.
i think the problem may be here: https://github.com/karmada-io/karmada/blob/master/pkg/scheduler/core/division_algorithm.go#L36

Accoding to this logic, in the above case, assignedReplicas == spec.Replicas ==>  the target clusters not changed ==> no need to reschedule

**Environment**:
- Karmada version: v1.0.1
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by dddddai** [comment]
I've just tested it

/assign

**Comment by RainbowMango** [comment]
@Garrybest @huone1

**Comment by huone1** [comment]
it is resolved in PR #1383

**Comment by dddddai** [comment]
Yeah I just took a look at #1383 and it should work
